ambry-network/src/main/java/com.github.ambry.network/Selector.java

```diff
@@ -455,6 +481,9 @@ private void close(SelectionKey key) {
       activeConnections.set(this.keyMap.size());
       try {
         transmission.close();
+        if (pendingHandshakeTransmissions.get(transmission.getConnectionId()) != null) {
+          pendingHandshakeTransmissions.remove(transmission.getConnectionId());
+        }
       } catch (IOException e) {
```

Comment:
yes, looks like we might have an issue. even keyMap and activeConnections looks to be affected. Lets discuss in person.